apps/web/src/components/tx-flow/flows/ExecuteBatch/ReviewBatch.tsx

```diff
@@ -118,6 +121,20 @@ export const ReviewBatch = ({ params }: { params: ExecuteBatchFlowProps }) => {
     )
   }
 
+  const [safeTx] = useAsync<SafeTransaction | undefined>(async () => {
+    const safeTx = multiSendTxs ? await createMultiSendCallOnlyTx(multiSendTxs) : undefined
+
+    if (safeTx) {
+      // For simulation purposes, we need to estimate gas even if the Safe version doesn't require it
+      const { safeTxGas } = await fetchRecommendedParams(safe.chainId, safe.address.value, safeTx.data)
+      safeTx.data.safeTxGas = safeTxGas
+    }
+
+    return safeTx
+  }, [multiSendTxs, safe.chainId, safe.address.value])
+
+  useSafeShieldForTxData(safeTx)
```

Comment:
Why do we do this in individual flows instead of some central place like the tx flow provider?